README.md
<!--
 * @Author: liuYang
 * @Description: 请填写描述信息
 * @Path: 引入路径
 * @Date: 2021-03-09 15:29:22
 * @LastEditors: liuYang
 * @LastEditTime: 2021-08-20 17:43:07
 * @MustParam: 必传参数
 * @OptionalParam: 选传参数
 * @EmitFunction: 函数
-->
# qishi_xxxx

Vue+Element-ui搭建的后台项目架子  新建项目请注意  @config文件配置和env文件

## Project setup
```
yarn install
```

### Compiles and hot-reloads for development
```
yarn serve

运行项目请注意env开发环境内环境配置  

# 后端请求地址
VUE_APP_SERVICE_HOST = "api"

# 请求图片的路径
VUE_APP_CAPTCHA_HOST = "http://192.168.200.212:7001"

# CAS域名 本地写自己本地IP
VUE_APP_CAS_HOST = ""

# 知识库 本地写自己本地IP
VUE_APP_WIKI_HOST = "http://192.168.3.95:7005"

# 热线 本地写自己本地IP
VUE_APP_HOT_LINE_HOST = "http://192.168.3.45:7005"

...

```

### Compiles and minifies for production
```
yarn build
```

### Lints and fixes files
```
yarn lint
```

### Customize configuration
See [Configuration Reference](https://cli.vuejs.org/config/).

### Vue 项目目录规范

    ├── dist                        // [生成] 打包目录
    ├── src                         // [必选] 开发目录
    │    ├── api                    // [生成] 后端接口
    │    ├── views                  // [必选] 页面组件，不允许有其他类型组件混入
    │    ├── components             // [必选] 业务组件必须写在这里
    │    ├── config                 // [可选] 配置目录
    │    ├── mock                   // [可选] mock 数据
    │    ├── test                   // [可选] 测试代码
    │    ├── utils                  // [可选] 工具库（用于一些函数方法之类的库）
    │    └── assets                 // [可选] 公共资源（被项目引用的经过webpack处理的资源）
    │       ├── css                 // [可选] css
    │       ├── js                  // [可选] js
    │       ├── img                 // [可选] 图片
    │       └── theme               // [可选] element主题色 [element主题配置](https://element.eleme.cn/#/zh-CN/theme)
    │    ├── store                  // [可选] 数据存储 vuex
    │    ├── route                  // [可选] 路由
    │    ├── style                  // [可选] 公共样式
    │    ├── App.vue                // [必选] 根组件
    │    └── main.(js|ts)           // [必选] 入口文件
    ├── public                      // [必选] 不会被webpack编译的资源
    │    ├── index.html             // [必选] 模板
    │    └── logo.png               // [可选] 项目 logo
    ├── docs                        // [可选] 文档
    │── .gitignore                  // [必选] git 忽略的文件
    │── .editorconfig               // [必选] 编译器配置
    │── .npmignore                  // [可选] 如果是 npm 包是必选
    │── jsconfig.json               // [可选] 用于 vscode 配置
    ├── README.md                   // [必选] 导读
    ├── package.json                // [必选] 大家都懂
    └── ......                      // [可选] 一些工具的配置，如果 babel.config.js 等

### components、views 具体职能划分
components 只写公共组件，页面附带组件写在 views 内

    └── src
        ├── views
        │    └── Home
        │         ├── index.vue
        |         └── components
        │           ├── Banner.vue
        │           └── Card.vue
        ├── components
        │    ├── Swiper.vue
        │    └── Button.vue
        ├── store                 // 对于较大的项目，建议按业务模块拆分管理
        │    ├── index.js
        │    ├── home.js
        │    └── mine.js
        ├── route                 // 对于较大的项目，建议按业务模块拆分管理
        │    ├── index.js
        │    ├── home.js
        │    └── mine.js
        └── assets                // 重复使用的公共资源放在顶层 assets 文件，避免重复定义
             ├── css
             ├── js
             ├── img
             └── iconfont

### 其他项目规范 

[小石前端项目规范和插件配合](http://gitlab.beijingxiaoshi.cn/qianduan/components/cooperation)

### 其他补充的文档 

[iframe打开子系统](http://gitlab.beijingxiaoshi.cn/qianduan/components/vue_init_cli/-/tree/master/doc)  
[热线系统文档](http://gitlab.beijingxiaoshi.cn/qianduan/components/vue_init_cli/-/blob/master/doc)  
[现有公共组件](http://gitlab.beijingxiaoshi.cn/qianduan/components/vue_init_cli/-/blob/master/doc/%E5%85%AC%E5%85%B1%E7%BB%84%E4%BB%B6%E3%80%81%E6%96%B9%E6%B3%95.md)  
[单点登录CAS端](http://gitlab.beijingxiaoshi.cn/qianduan/components/vue_init_cli/-/blob/master/doc/%E5%8D%95%E7%82%B9%E7%99%BB%E5%BD%95CAS%E5%A4%84%E7%90%86.md)  
[单点登录子系统处理](http://gitlab.beijingxiaoshi.cn/qianduan/components/vue_init_cli/-/blob/master/doc/%E5%8D%95%E7%82%B9%E7%99%BB%E5%BD%95%E5%AD%90%E7%B3%BB%E7%BB%9F%E5%A4%84%E7%90%86.md)  
[路由架构处理](http://gitlab.beijingxiaoshi.cn/qianduan/components/vue_init_cli/-/blob/master/doc/%E8%B7%AF%E7%94%B1%E6%9E%B6%E6%9E%84%E5%A4%84%E7%90%86.md) 


CHANGELOG.md
# 0.1.0 (2021-08-23)

### ⚙ 组件变化
范围|描述|commitId
--|--|--
|  影响整个项目 | 删除提醒Dialog新增插槽 | 96fb600|


### ➕ 添加依赖
范围|描述|commitId
--|--|--
|  无影响 | 更新 git commitlint 依赖 | 87c7af6|
|  影响整个项目 | 修改错误版本号 | ead6753|
|  global | 移除了vue-cli-plugin-commitlint插件  引入了ly-tool-git-commitlint插件 | 3f62a20|
|  noproblem | 更新依赖包 | 26e69bb|


### ➖ 删除依赖
范围|描述|commitId
--|--|--
|  无影响 | 修改错误版本号 | 2e73f88|
|  影响整个项目 | ceshi | 773c96b|


### 🌟 新功能
范围|描述|commitId
--|--|--
|  影响整个项目 | 处理iframe打开其他系统 | f196247|
|  no select | 补充导航条样式 | 5495933|
|  no select | 处理登录页面 | 662d725|
|  no select | 处理请求跨域 | 5dc683b|
|  no select | 处理通知中心 | 5c77ab0|
|  no select | 面板样式优化 | 04b7cc5|
|  no select | 软话条组件 | 987d2d2|
|  no select | 新增导航用户信息组件 | bd61f2d|
|  no select | 修改菜单样式并组件化 | 81f960f|
|  no select | 样式修改 | 1d2f4b3|
|  no select | UI改动 | 88bbbec|


### 🎉 init
范围|描述|commitId
--|--|--
|  无影响 | init | d2b0426|
|  影响整个项目 | 持续集成基础cli | 556c972|
|  影响整个项目 | 完善init 路由跳转等初始化 | 4f97509|
|  影响整个项目 | 修改项目名称 | cb83c41|
|  影响整个项目 | 优化依赖 | a169e72|


### 🎨 代码样式
范围|描述|commitId
--|--|--
|  no select | 规范格式化文件头注释 | 7798054|
|  更新iconfont | 更新iconfont | bc0da42|
|  no select | 代码优化 | 34b40a3|
|  no select | 删除无用代码 | 16bf4bc|
|  no select | 删除无用的代码 | 997b1a0|
|  更改一些基础配置 | 一些基础配置 | c5d267c|
|  no select | 文件名称修改 | 539d8fd|


### 🏠 构建系统
范围|描述|commitId
--|--|--
|  影响整个项目 | 集成优化 | 9ba2338|
|  影响整个项目 | 新增依赖的处理 | b46a7dc|


### 🐛 Bug修复
范围|描述|commitId
--|--|--
|  影响整个项目 | 处理浏览器指纹 | 5ad4339|
|  影响整个项目 | 处理权限缓存读取错误的bug | 49df99c|
|  影响整个项目 | 处理树形数据的算fan | b25951d|
|  no select | 修复页面可以无限缩小的bug | d029100|


### 💄 UI变更
范围|描述|commitId
--|--|--
|  UI 界面 | 处理导航条UI | 90720b4|
|  UI 界面 | 处理UI界面 | 21c0cc8|


### 📝 文档
范围|描述|commitId
--|--|--
|  无影响 | 补充公共README | 310cd91|
|  影响整个项目 | 补充文档 | 6697d08|
|  影响整个项目 | 文档补充 | 4c741f9|
|  影响整个项目 | 文档补充 | bed891a|
|  noproblem | 更新文档 | ad70716|
|  no select | 修改文档 | 66c2bd6|


### 🔙 撤销上一次的提交
范围|描述|commitId
--|--|--
|  no select | 上传错误 | 6abe97b|


### 🔨 代码重构
范围|描述|commitId
--|--|--
|  影响整个项目 | 1.重构utils/storage 里clearAllStorage | d9f78cb|
|  影响整个项目 | 处理iframe打开 | f016199|
|  影响整个项目 | 同步近期对架构的改动 | 8e456aa|
|  影响整个项目 | 优化一些东西 | b669d0c|
|  no select | 修改一点无影响的样式 | 390c03f|
|  no select | 优化组件化程度 | e0ff869|


### 🚀 性能优化
范围|描述|commitId
--|--|--
|  优化面板页面 | 优化面板页面 | c614c34|
|  no select | 优化变量名 | 38eb8cd|


范围|描述|commitId
--|--|--
|  no select | Update .env.production | 99b36d9|
|  no select | Update .env | 2f68710|
|  no select | 样式优化 | 552187f|
|  no select | 处理错误情况 | 08ae87a|
|  no select | 处理样式 | ab138dd|
|  no select | 优化样式 | 6ae2b82|
|  no select | 处理样式引入 | 53dc890|
|  no select | 优化样式 | d9878db|
|  no select | 优化代码和目录结构 | 4677033|
|  no select | 优化代码 | 8115665|
|  no select | 删除无用代码 | c8e552a|
|  no select | 处理UI | 18f7ae6|
|  no select | 修改样式 | 2d28668|
|  no select | axios封装优化 | 996cf2d|
|  no select | 补充文档 | df0f820|
|  no select | 优化书写 | df8167b|
|  no select | 新增gitcommit 依赖 | 44c6262|
|  no select | 新增Git commit | 0418f1a|
|  no select | 新增编辑器配置 | d8d9b75|
|  no select | 更新readme | a7cbefb|
|  no select | 优化init | 2b99d60|
|  no select | init | b0c1b92|
|  no select | 处理问题 | 50e5722|
|  no select | init | c8901d3|
|  no select | init | de3a6d1|
|  no select | init | 0a18517|
|  no select | 初始化项目 | a505087|
|  no select | init | e3f1487|
|  no select | 修改eslint | 90cae56|
|  no select | 严格ESLINT | 766acc6|
|  no select | init | 0c679ab|


### chore
范围|描述|commitId
--|--|--
|  no select | 更改部分webpack配置 | 43d9c1f|
|  no select | 添加 本地环境变量 | 95282e0|
|  no select | 新增axios包 | 9777087|
|  no select | 新增CAS环境变量 | ef7aa45|
|  no select | 修改错误的目录结构 | aa1941c|
|  no select | 重新打开每次启动项目自动打开页面 | 4a3dfff|


### ci
范围|描述|commitId
--|--|--
|  no select | 更新一些东西 | dd67830|



vue.config.js
/*
 * @Author: liuYang
 * @description: 请填写描述信息
 * @Path:  引入路径
 * @Date: 2021-03-09 15:33:35
 * @LastEditors: liuYang
 * @LastEditTime: 2021-04-19 17:38:12
 * @MustParam:  必传参数
 * @OptionalParam:  选传参数
 * @EmitFunction:  函数
 */
const path = require('path')
const IS_PROD = ['production', 'prod', 'pro', 'test'].includes(process.env.NODE_ENV)
const resolve = (dir) => path.join(__dirname, dir)
// gzip压缩
const CompressionWebpackPlugin = require('compression-webpack-plugin')

const productionGzipExtensions = /\.(js|css|json|txt|html|ico|svg)(\?.*)?$/i
// 添加打包分析
// const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin

module.exports = {
  // 线上资源读取地址
  publicPath: process.env.VUE_APP_PUBLIC_PATH,
  // outputDir: "",
  lintOnSave: true,
  // 设为false打包时不生成.map文件
  productionSourceMap: false,
  // 这里写你调用接口的基础路径，来解决跨域，如果设置了代理，那你本地开发环境的axios的baseUrl要写为 '' ，即空字符串
  devServer: {
    // host: "192.168.31.246",
    open: true,
    port: 7001,
    hotOnly: false,
    disableHostCheck: true,
    proxy: {
      //在此配置跨域
      '/api/': {
        target: 'http://192.168.200.212:7001', //后端的接口地址
        secure: false,
        changeOrigin: true,
        pathRewrite: {
          '^/api': '' //路径重写
        }
      }
    }
  },
  chainWebpack: (config) => {
    // 修复HMR
    config.resolve.symlinks(true)

    config.resolve.alias
      .set('@', resolve('src'))
      .set('@components', resolve('src/components'))
      .set('@views', resolve('src/views'))
      .set('@api', resolve('src/api/modules'))
      .set('@utils', resolve('src/utils'))
      .set('@js', resolve('src/assets/js'))
      .set('@css', resolve('src/assets/css'))
      .set('@img', resolve('src/assets/img'))
      .set('@config', resolve('src/config'))
      .set('@router', resolve('src/router'))
      .set('@store', resolve('src/store'))
      .set('@mixins', resolve('src/mixins'))
      .set('@layout', resolve('src/layout'))
    config.module
      .rule('pug')
      .test(/\.pug$/)
      .use('pug-html-loader')
      .loader('pug-html-loader')
      .end()
    // if (IS_PROD) {
    //   // 打包生成打包分析
    //   config.plugin('webpack-report').use(BundleAnalyzerPlugin, [
    //     {
    //       analyzerMode: 'static'
    //     }
    //   ])
    // }
  },
  // gzip
  configureWebpack: (config) => {
    const plugins = []
    if (IS_PROD) {
      plugins.push(
        new CompressionWebpackPlugin({
          filename: '[path].gz[query]',
          algorithm: 'gzip',
          test: productionGzipExtensions,
          threshold: 10240,
          minRatio: 0.8
        })
      )
    }
    // End 生成 gzip 压缩文件
    config.plugins = [...config.plugins, ...plugins]
  }
}
